### ISSUE TITLE
[BUG] cachedb_local - Failed to process a binary packet

### ISSUE THREAD

**Issue by vtzan** [OpenSIPS/opensips#2092]
**OpenSIPS version you are running**

```
version: opensips 3.0.2 (x86_64/linux)
flags: STATS: On, DISABLE_NAGLE, USE_MCAST, SHM_MMAP, PKG_MALLOC, Q_MALLOC, F_MALLOC, HP_MALLOC, DBG_MALLOC, FAST_LOCK-ADAPTIVE_WAIT
ADAPTIVE_WAIT_LOOPS=1024, MAX_RECV_BUFFER_SIZE 262144, MAX_LISTEN 16, MAX_URI_SIZE 1024, BUF_SIZE 65535
poll method support: poll, epoll, sigio_rt, select.
git revision: 26bcfb77e
main.c compiled on 16:06:42 Apr 27 2020 with gcc 8
```

**Describe the bug**
I am using cachedb_local to replicate cache key data between 2 servers. The replication
procedure seems to send correctly the data to the other side but it seems that the side that receives the replication packets can't process them correctly.

opensips.cfg
```
loadmodule "cachedb_local.so"
modparam("cachedb_local", "cache_clean_period", 300)
modparam("cachedb_local", "cachedb_url", "local:user_profile:///collection1")
modparam("cachedb_local", "cache_collections", "collection1;")
modparam("cachedb_local", "cluster_id", MY_CLUSTER_ID)
modparam("cachedb_local", "cluster_persistency", "sync-from-cluster")
```

**Relevant System Logs**
https://pastebin.com/cqG694wD

**OS/environment information**
 - Operating System: Debian GNU/Linux 10 (buster)
 - OpenSIPS installation: git
 - other relevant information:

Thank you in advance for your support

Best Regards
Vasilios Tzanoudakis

**Comment by fgast** [comment]
Hi @vtzan Thanks for reporting this
I have found the issue in your case & i am preparing a fix for this.

**Comment by fgast** [comment]
@vtzan  You can apply the PR from  https://github.com/OpenSIPS/opensips/pull/2196 to your sources (this should work for 3.0.x too)  and rebuild opensips - this should solve your problem.

**Comment by vtzan** [comment]
@fgast thank you for your fix I will try it on 3.1 which is the main branch that I use right now and let you know. Thanks

**Comment by rvlad-patrascu** [comment]
Hi @vtzan ,

You can now try the proper fix from the public 3.0.
